apps/test/unit/componentLibrary/ToggleTest.jsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import sinon from 'sinon';
+
+import {expect} from '../../util/reconfiguredChai';
+
+import Toggle from '@cdo/apps/componentLibrary/toggle';
+describe('Design System - Toggle', () => {
+  it('Toggle - renders with correct label', () => {
+    render(
+      <Toggle name="test-toggle" value="test-toggle" label="Toggle label" />
+    );
+
+    const toggle = screen.getByDisplayValue('test-toggle');
+    expect(toggle).to.exist;
+    expect(screen.getByText('Toggle label')).to.exist;
+  });
+
+  it('Toggle - changes checked state on click', async () => {
+    const user = userEvent.setup();
+    const spyOnChange = sinon.spy();
+
+    let checked = false;
+    const onChange = () => {
+      checked = !checked;
+      spyOnChange(checked);
+    };
+
+    // Initial render
+    const {rerender} = render(
+      <Toggle
+        name="test-toggle"
+        value="test-toggle"
+        label="Toggle label"
+        checked={checked}
+        onChange={onChange}
+      />
+    );
+
+    let toggle = screen.getByDisplayValue('test-toggle');
+
+    expect(toggle).to.exist;
+    expect(toggle.checked).to.be.false;
+    expect(toggle.disabled).to.be.false;
+
+    await user.click(toggle);
+
+    // Re-render after user's first click
+    rerender(
+      <Toggle
+        name="test-toggle"
+        value="test-toggle"
+        label="Toggle label"
+        checked={checked}
+        onChange={onChange}
+      />
+    );
+
+    toggle = screen.getByDisplayValue('test-toggle');
+
+    expect(spyOnChange).to.have.been.calledOnce;
+    expect(spyOnChange).to.have.been.calledWith(true);
+    expect(toggle.checked).to.be.true;
+    expect(toggle.disabled).to.be.false;
+
+    await user.click(toggle);
```

Comment:
New to RTL -- [docs here say](https://testing-library.com/docs/ecosystem-user-event/#api):

> Note: All userEvent methods are synchronous with one exception: when delay option used with userEvent.type as described below. We also discourage using userEvent inside before/after blocks at all, for important reasons described in ["Avoid Nesting When You're Testing"](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing).

Why do we need async here then?

Reply:
The link you've shared here references user-event v13, which is no longer maintained. 
We're using  `"@testing-library/user-event": "^14.4.3",`, you can [find docs for it here](https://testing-library.com/docs/user-event/intro) 

Re async: Since v14 user-event APIs always return promise(this was one of breaking changes actually) [changelog link](https://github.com/testing-library/user-event/releases/tag/v14.0.0)

I've also google. a little bit and here's a quote from the [article about migrating from v13 to v14](https://medium.com/software-fast-radius/upgrading-react-testing-librarys-user-event-to-version-14-d3fee658247):
 `Also notice we had to add an async in the test setup options, and are adding an await before user.click. Previously when calling userEvent.click, it would automatically wrap an await around the action. Not anymore. Now you have to be explicit about calling await. In my experience upgrading, although there may be a few rare cases where an await isn’t needed, you can pretty much always assume you’ll need it and default to using it.`

Reply:
Ack sorry, always forget to check the version when looking at docs. Thanks for the updated docs / explanation!